home/models.py

```diff
@@ -17,34 +19,36 @@ def __str__(self):
         return f"{self.name} | {self.email}"
 
     class Meta:
-        ordering = ('-created',)
-        constraints = [
-            models.UniqueConstraint(fields=['email', 'device_id'], name='Unique user + device')
-        ]
+        ordering = ("-created",)
+
 
 # Event model
 class DailyWalk(models.Model):
+    # Identifier
+    event_id = models.CharField(max_length=250, primary_key=True)
+    # Walk meta
     date = models.DateField()
     steps = models.IntegerField()
-    user = models.manufacturer = models.ForeignKey('AppUser',
-                                                   on_delete=models.CASCADE)
+    appuser = models.manufacturer = models.ForeignKey("AppUser", on_delete=models.CASCADE)
 
     def __str__(self):
-        return f"{self.user} | {self.date}"
+        return f"{self.appuser} | {self.date}"
 
     class Meta:
-        ordering = ('-date',)
+        ordering = ("-date",)
+
 
 # Event model
 class IntentionalWalk(models.Model):
+    event_id = models.CharField(max_length=250, primary_key=True)
+    # Walk meta
     start = models.DateTimeField()
     end = models.DateTimeField()
     steps = models.IntegerField()
```

Comment:
Same comment as above, re: saving distance

Reply:
So, working on the recorded IntentionalWalk, I think we'll also need to store another field that is "pause time".

In the UI, it is possible to "pause" recording. What I do in this case in the app is I start keeping track of how long it is in the pause state, and when the user resumes recording, I store that duration of time (in seconds) as an additional attribute called "pause". If the user pauses/resumes multiple times during a walk, I just aggregate that collective pause time together. That way, after the user stops and saves, then the "elapsed time" is ((end - start) - pause)).

Reply:
Sounds good! Will add this in

Reply:
I will have an integer field for this and units would be seconds I'm guessing?

Reply:
So one thing is, the app install is lost, then the pause times are also lost. My guess is that isn't a big deal and we don't need the pause-resume list in the backend too

Reply:
I'm not keeping track of each individual pause/resume segment in the app either. Seemed like overkill...?

Reply:
Yeah... we can add that in later versions if needed or if the product design folks think they might need it 🤷‍♂